tools/ports/__init__.py

```diff
@@ -63,6 +64,26 @@ def load_port(name):
       utils.exit_with_error('duplicate port variant: %s' % variant)
     port_variants[variant] = (port.name, extra_settings)
 
+  validate_port(port)
+
+
+def load_port_by_name(name):
+  port = __import__(name, globals(), level=1, fromlist=[None])
+  init_port(name, port)
+
+
+def load_port_by_path(path):
+  name = os.path.splitext(os.path.basename(path))[0]
+  if name in ports_by_name:
+    utils.exit_with_error(f'port path [{path}] is invalid: duplicate port name {name}')
+  module_name = f'tools.ports.{name}'
+  spec = importlib.util.spec_from_file_location(module_name, path)
+  port = importlib.util.module_from_spec(spec)
+  sys.modules[module_name] = port
+  spec.loader.exec_module(port)
```

Comment:
Its kind of crazy how much arcane python innards we need to do here just to load some code from a file! :)

Reply:
I know! I suppose it is right. I have to admit I did a search for this code. If you think there is a better way, I would be more than happy to change it.